nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/QueryComparator.java

```diff
@@ -0,0 +1,35 @@
+package io.github.nucleuspowered.nucleus.api.query;
+
+public enum QueryComparator {
+
+    /**
+     * General comparators.
+     */
+    LESS_THAN("<", 1),
+    LESS_THAN_OR_EQUAL_TO("<=", 1),
+    EQUALS("=", 1),
+    GREATER_THAN_OR_EQUAL_TO(">=", 1),
+    GREATER_THAN(">", 1),
+
+    /**
+     * Compares if a timestamp is between a certain range of timestamps.
+     */
+    BETWEEN("BETWEEN", 2);
```

Comment:
The API should be _completely_ agnostic to the implementation. If I was to introduce a, for the sake of argument, MongoDB based backend later, these tokens might be irrelevant and updating this would break the API. Most of this is an implementation detail, and therefore do not belong in the API layer, and should be put somewhere in the implementation.

The API package should _only_ contain enough so that a consumer of the API can interact, whilst we can make changes later. This should either be a simple API, or a set of keys that implement a common interface that we can implement later.